.github/workflows/create-cache.yml

```diff
@@ -0,0 +1,291 @@
+## Github workflow to create reusable caches
```

Comment:
I'd like an explanation of the cache here if possible, because I don't understand fully. It's to prevent building the same sources repeatedly?

Reply:
correct. i chose using the cache very specifically for these workflows (vs using an artifact like we do currently). 

the main reason is that the caches are *fast* to restore from, and will persist across workflow runs (as long as the cleanup workflow hasn't deleted it) - making it possible to easily re-run a failed workflow without having to recreate the cache. 

secondly, with a cache - it's available to every workflow that runs vs only the local workflow, so we can split apart different jobs (i.e. atlas tests/bitcoin tests) to better parallelize them. 

as noted above though, i have seen a few cases where the cache fails to restore (which is why i added a timeout), resulting in the workflow failing due to the cache, and not a failed test etc. it's rare, but it does happen.

Reply:
the caveat with re-using the cache obviously is the key. for the bitcoin binary cache, it's static (although i may see about using a version in the cache name). 
as long as the commit sha matches for a PR's cache, you can reuse an existing cache (if there has been a commit, however, a new cache is needed)